src/webapp/gradebookServer.js

```diff
@@ -280,6 +280,57 @@ app.get('/sections', function(request, response) {
    });
 });
 
+
+//Returns an array of dates for a given section id.  Omits dates when classes are
+// not held.
+app.get('/sectionschedule', function(request, response) {
+   //Decrypt the password recieved from the client.  This is a temporary development
+   //feature, since we don't have ssl set up yet
+   var passwordText = sjcl.decrypt(superSecret, JSON.parse(request.query.password));
+
+   //Connnection parameters for the Postgres client recieved in the request
+   var config = createConnectionParams(request.query.user, request.query.database,
+      passwordText, request.query.host, request.query.port);
+
+   var sectionID = request.query.sectionID;
+
+   var queryText = 'SELECT date FROM significantDate WHERE classesheld = false;';
+   var queryParams = [];
+   var closedDates = [];
+
+   executeQuery(response, config, queryText, queryParams, function(result) {
+      for(row in result.rows) {
+         closedDates.push(
+            {
+               "date": result.rows[row].date
+            }
+         );
+      }
+   });
+
+   queryText = 'SELECT getScheduleDates($1);';
+   queryParams = [sectionID];
+
+   executeQuery(response, config, queryText, queryParams, function(result) {
+      var classDates = [];
+      for(row in result.rows) {
+         if(!closedDates.includes(result.rows[row].date))
+         {
+            classDates.push(
```

Comment:
I recommend condensing this and the earlier query parse loop (L303-L307) to just:

```js
classDates.push(result.rows[row].date);
```

This way, an array of dates as strings is returned. (Currently, it returns an array of objects.)